src/ast/passes/codegen_llvm.cpp

```diff
@@ -1415,7 +1416,20 @@ void CodegenLLVM::visit(Offsetof &ofof)
 void CodegenLLVM::visit(Map &map)
 {
   auto [key, scoped_key_deleter] = getMapKey(map);
-  Value *value = b_.CreateMapLookupElem(ctx_, map, key, map.loc);
+
+  auto map_info = bpftrace_.resources.maps_info.find(map.ident);
+  if (map_info == bpftrace_.resources.maps_info.end()) {
+    LOG(BUG) << "map name: \"" << map.ident << "\" not found";
+  }
+
+  const auto &val_type = map_info->second.value_type;
+  Value *value;
+  if (val_type.IsCountTy() || val_type.IsSumTy()) {
```

Comment:
Can we add a helper method (maybe in types.h?) that groups cast-able map types? It'll help make things more clear across semantic analysis && codegen